Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h

```diff
@@ -177,8 +177,105 @@
+  #if !HAS_RESOLUTION
+    #define TOUCH_UI_800x480
+    #define TOUCH_UI_800x480_GENERIC    // use more common timing parameters as the original set
+  #endif
+  #ifndef FTDI_API_LEVEL
+    #define FTDI_API_LEVEL                810
+  #endif
+  namespace FTDI {
+    IS_FT810
+    constexpr bool Use_Crystal            = true; // 0 = use internal oscillator, 1 = module has a crystal populated
+    constexpr bool GPIO_0_Audio_Enable    = false;
+    constexpr bool GPIO_1_Audio_Shutdown  = false;
+    #define USE_GT911 // this display uses an alternative touch-controller and we need to tell the init function to switch
+    constexpr uint8_t Swizzle             = 0;
+    constexpr uint8_t CSpread             = 1;
+    constexpr uint8_t Pclkpol             = 1;
+    constexpr uint16_t touch_threshold    = 1200; /* touch-sensitivity */
+
+    constexpr uint32_t default_transform_a  =  0x000109E4;
+    constexpr uint32_t default_transform_b  =  0x000007A6;
+    constexpr uint32_t default_transform_c  =  0xFFEC1EBA;
+    constexpr uint32_t default_transform_d  =  0x0000072C;
+    constexpr uint32_t default_transform_e  =  0x0001096A;
+    constexpr uint32_t default_transform_f  =  0xFFF469CF;
+  }
+
+/**
+ * Settings for EVE2-50G - Matrix Orbital 5.0" 800x480, FT813
+ * https://www.matrixorbital.com/ftdi-eve/eve-bt815-bt816/eve3-50g
+ * use for example with: https://github.com/RudolphRiedel/EVE_display-adapter
+ */
+#elif defined(LCD_EVE2_50G)
+  #if !HAS_RESOLUTION
+    #define TOUCH_UI_800x480
+    #define TOUCH_UI_800x480_GENERIC    // use more common timing parameters as the original set
+  #endif
+  #ifndef FTDI_API_LEVEL
+    #define FTDI_API_LEVEL                810
+  #endif
+  namespace FTDI {
+    IS_FT810
+    constexpr bool Use_Crystal            = false; // 0 = use internal oscillator, 1 = module has a crystal populated
+    constexpr bool GPIO_0_Audio_Enable    = false;
+    constexpr bool GPIO_1_Audio_Shutdown  = false;
+    #define PATCH_GT911 // this display uses an alternative touch-controller and we need to tell the init function to patch the FT813 for it
+    constexpr uint8_t Pclkpol             = 1;
+    constexpr uint8_t Swizzle             = 0;
+    constexpr uint8_t CSpread             = 1;
+    constexpr uint16_t touch_threshold    = 1200; /* touch-sensitivity */
+
+    constexpr uint32_t default_transform_a  =  0x000109E4;
+    constexpr uint32_t default_transform_b  =  0x000007A6;
+    constexpr uint32_t default_transform_c  =  0xFFEC1EBA;
+    constexpr uint32_t default_transform_d  =  0x0000072C;
+    constexpr uint32_t default_transform_e  =  0x0001096A;
+    constexpr uint32_t default_transform_f  =  0xFFF469CF;
+  }
+
 #else
 
   #error "Unknown or no TOUCH_UI_FTDI_EVE board specified. To add a new board, modify this file."
 
 #endif
+
+
+/* this data is used to patch FT813 displays that use a GT911 as a touch-controller */
+#ifdef PATCH_GT911
+  constexpr PROGMEM unsigned char GT911_data[] = {
```

Comment:
What does this do? Where did it com from?

Reply:
This is from Bridgeteks application note "AN_336 FT8xx - Selecting an LCD Display" and it applies a patch to the FT813 in order to make it support Goodix GT911 touch controllers.
Out of the box the FT813 mostly supports FocalTech touch controllers, the IQS550 from Azoteq and the HY4613 from Hycon.

The EVE2-xxG modules from Matrix Orbital need this patch and the Crystalfontz CFAF800480E0-050.
BT815 like in the EVE3 series do support the GT911 natively but have to configured for it.

This is why this is inside an #ifdef, to usually not add this to the binary.

Reply:
Would you mind adding this or a similar explanation to the comment in the code? I can imagine someone looking at it later and asking the same questions I did.

I looked up that AN you mentioned. I must say I was a little surprised to find big binary blobs like that in a PDF :)

Reply:
Well, there are comments in to explain what this is, not only line 252 but also lines 199 and 231.
And line 1102ff of commands.pp